app/services/course/assessment/submission/update_service.rb

```diff
@@ -167,15 +167,16 @@ def auto_grade(answer)
     end
   end
 
-  def grade_and_reattempt_answer(answer)
-    # The transaction is to make sure that auto grading and job are present when the answer is in
-    # the submitted state.
+  def reattempt_and_grade_answer(answer)
+    # The transaction is to make sure that the new attempt, auto grading and job are present when
+    # the current answer is submitted.
     answer.class.transaction do
-      answer.finalise! if answer.attempting?
-      # Only save if answer is graded in another server
-      answer.save! unless answer.grade_inline?
-      answer.auto_grade!(redirect_to_path: nil,
-                         reattempt: true, reduce_priority: false)
+      last_answer = answer.submission.answers.select { |ans| ans.question_id == answer.question_id }.last
+      last_answer.destroy! if last_answer&.auto_grading&.job&.errored?
```

Comment:
Is it possible that last_answer is current answer?

Reply:
On the first answer evaluation, but then it wouldn't have an auto_grading so it won't get destroyed.

For manually graded submissions which were immediately submitted, the `Evaluate Answers` button doesn't go through this code path.

Reply:
What if:
1. submit the submission
2. current answer will be evaluated and let's say the job gets errored
3. unsubmit submission

In the above case what will be the last answer here?

Reply:
OK I see, Then should be fine. would be good to have a test to guard that.